src/oic/oauth2/message.py

```diff
@@ -439,6 +445,15 @@ def _add_value(self, skey, vtyp, key, val, _deser, null_allowed):
                         val = _deser(val, sformat="dict")
                     except Exception as exc:
                         raise DecodeError(ERRTXT % (key, exc))
+                elif vtyp == int:
+                    try:
+                        self._dict[skey] = int(val)
+                    except (ValueError, TypeError):
```

Comment:
I'm not ignoring TypeError. I'm just raising a ValueError if for some reason 'val' can not be 
cast to an int.